### PR TITLE
init CEL validation

### PR DESCRIPTION
**Pull request by zirain**
xref: https://github.com/envoyproxy/gateway/issues/1964

### CODE REVIEW

File: api/v1alpha1/envoyproxy_types.go [hunk 1]
```diff
@@ -97,6 +97,7 @@ type EnvoyProxyProvider struct {
 	// optional auxiliary control planes. Supported types are "Kubernetes".
 	//
 	// +unionDiscriminator
+	// +kubebuilder:validation:XValidation:message="type must be one of ['Kubernetes']",rule="self == 'Kubernetes'"
```
Comment (arkodg) [inline api/v1alpha1/envoyproxy_types.go:100]: is this needed ?
it already exists https://github.com/envoyproxy/gateway/blob/bdb0191dec04788a1b8e47b5a66f0cc99ef29afe/api/v1alpha1/shared_types.go#L36

File: tools/make/kube.mk [hunk 2]
```diff
@@ -210,3 +210,12 @@ generate-egctl-releases: ## Generate egctl releases
 	curl -sSL https://github.com/envoyproxy/gateway/releases/download/latest/egctl_latest_darwin_arm64.tar.gz -o $(OUTPUT_DIR)/egctl_$(TAG)_darwin_arm64.tar.gz
 	curl -sSL https://github.com/envoyproxy/gateway/releases/download/latest/egctl_latest_linux_amd64.tar.gz -o $(OUTPUT_DIR)/egctl_$(TAG)_linux_amd64.tar.gz
 	curl -sSL https://github.com/envoyproxy/gateway/releases/download/latest/egctl_latest_linux_arm64.tar.gz -o $(OUTPUT_DIR)/egctl_$(TAG)_linux_arm64.tar.gz
+
+.PHONY: validation
```
Comment (arkodg) [inline tools/make/kube.mk:214]: imo we should reuse `make test` or `make kube-test` for this and not create another target

File: .github/workflows/build_and_test.yaml [hunk 3]
```diff
@@ -2,172 +2,167 @@ name: Build and Test
 on:
   push:
     branches:
-    - "main"
-    - "release/v*"
+      - "main"
```
Comment (arkodg) [inline .github/workflows/build_and_test.yaml:5]: can we undo all changes from this file  ?
the existing indentation seems consistent with our code base

File: test/validation/main_test.go [hunk 4]
```diff
@@ -0,0 +1,66 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+//go:build validation
```
Comment (arkodg) [inline test/validation/main_test.go:6]: can we call this `cel-validation` ?

File: .github/workflows/build_and_test.yaml (hunk 3 above)
Comment (zirain) [reply .github/workflows/build_and_test.yaml:5]: that won't pass the yaml lint.

File: test/validation/main_test.go (hunk 4 above)
Comment (zirain) [reply test/validation/main_test.go]: done

File: .github/workflows/build_and_test.yaml (hunk 3 above)
Comment (arkodg) [reply .github/workflows/build_and_test.yaml:5]: Not seeing any issues when Im running yamllint

Comment (zirain) [reply .github/workflows/build_and_test.yaml:5]: you didn't change it?

Comment (zirain) [reply .github/workflows/build_and_test.yaml]: updated

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Review by Xunzhuo — APPROVED** [review]
LGTM, thanks.

**Pull request merged**